### PR TITLE
feat(bottlecap): add EventBridge inferred spans

### PR DESCRIPTION
**Pull request by alexgallotta**
- add event bridge
image

- change `api_gateway` -> `api-gateway` metadata as in https://github.com/DataDog/datadog-agent/blob/main/pkg/serverless/invocationlifecycle/constants.go#L45
- use `CARGO_MANIFEST_DIR` in test dir
- use host network to build bottlecap to speed up build and avoid timeouts

# Notes

Not doing EB-SQS or EB-SNS yet

### CODE REVIEW

File: bottlecap/tests/payloads/event_bridge_span.json
Comment (duncanista) [inline bottlecap/tests/payloads/event_bridge_span.json:1]: Create another folder under `tests`, this is technically not a payload, but a span, we could have `test/spans` or `test/traces` whichever route we are aiming for

File: bottlecap/tests/payloads/event_bridge.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "account": "601427279990",
+  "time": "2022-01-24T16:00:10Z",
+  "region": "eu-west-1",
+  "resources": [],
+  "detail": {
+    "hello": "there",
+    "_datadog": {
+      "x-datadog-trace-id": "5827606813695714842",
+      "x-datadog-parent-id": "4726693487091824375",
+      "x-datadog-sampled": "1",
```
Comment (duncanista) [inline bottlecap/tests/payloads/event_bridge.json:15]: We can remove the `x-datadog-sampled` here

Comment (alexgallotta) [reply bottlecap/tests/payloads/event_bridge.json:15]: is it an issue leaving it there? I would rather have the parser robust if this field might appear from time to time

File: bottlecap/src/lifecycle/invocation/span_inferrer.rs [hunk 2]
```diff
@@ -7,6 +7,7 @@ use tracing::debug;
 
 use crate::config::AwsConfig;
 
+use crate::lifecycle::invocation::triggers::event_bridge_event::EventBridgeEvent;
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/span_inferrer.rs:10]: Add under triggers

File: bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            }
+        }
+    }
+
+    fn is_match(payload: &Value) -> bool {
+        payload.get("detail-type").is_some()
+            && payload
+                .get("source")
+                .and_then(Value::as_str)
+                .map_or(false, |s| s != "aws.events")
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs:45]: Added this missed check

File: bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    fn get_arn(&self, _region: &str) -> String {
+        self.source.clone()
+    }
+
+    fn get_carrier(&self) -> HashMap<String, String> {
+        if let Ok(detail) = serde_json::from_value::<HashMap<String, Value>>(self.detail.clone()) {
+            if let Some(carrier) = detail.get(DATADOG_CARRIER_KEY) {
+                return serde_json::from_value::<HashMap<String, String>>(carrier.clone())
+                    .unwrap_or_default();
+            }
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs:86]: Updated the carrier code here, I really liked how we expect the detail to be a `Value`, might be really important to do this in every other field which could be a carrier

**Review by duncanista — APPROVED** [review]

**Comment by duncanista** [comment]
Actually, before merging, we might want to use the work @nhulston did to get an accurate time for EventBridge

File: bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+        let carrier = self.get_carrier();
+        let resource_name = carrier
+            .get(DATADOG_RESOURCE_NAME_KEY)
+            .unwrap_or(&self.source)
+            .clone();
+        let start_time = carrier
+            .get(DATADOG_START_TIME_KEY)
+            .and_then(|s| s.parse::<f64>().ok())
+            .map_or(start_time_seconds, |s| (s * MS_TO_NS) as i64);
```
Comment (duncanista) [inline bottlecap/src/lifecycle/invocation/triggers/event_bridge_event.rs:67]: Added resource name and start time accurate resolution when available

**Pull request merged**